lib/dao.js

```diff
@@ -757,12 +757,12 @@ DataAccessObject.findOrCreate = function findOrCreate(query, data, options, cb)
 
         // options has precedence on model-setting
         if (options.validate === false) {
-          _findOrCreate(query, data, obj);
+          return _findOrCreate(query, data, obj);
```

Comment:
I believe this was a bug; irrespective of whether `validate` is false or ... it was calling `_findOrCreate` after this block if the `obj` is valid; here was the block of code which was getting called:

```
        // validation required
        obj.isValid(function (valid) {
          if (valid) {
            _findOrCreate(query, data, obj);
          } else {
            cb(new ValidationError(obj), obj);
          }
        }, data);
```